src/extension-runners/chromium.js

```diff
@@ -127,8 +127,58 @@ export class ChromiumExtensionRunner {
       chromeFlags.push(...this.params.args);
     }
 
-    if (this.params.chromiumProfile) {
-      chromeFlags.push(`--user-data-dir=${this.params.chromiumProfile}`);
+    function isUserDataDir(dirCandidate: string): boolean {
+      const localStateCandidate = path.join(dirCandidate, 'Local State');
+      const defaultCandidate = path.join(dirCandidate, 'Default');
+      // Local State and Default are typical for the user-data-dir
+      return fs.pathExistsSync(localStateCandidate)
+        && fs.pathExistsSync(defaultCandidate);
+
+    }
+
+    function isProfileDir(dirCandidate: string): boolean {
+      const securePreferencesCandidate = path.join(
+        dirCandidate,
+        'Secure Preferences');
+      //Secure Preferences is typical for a profile dir
+      return fs.pathExistsSync(securePreferencesCandidate);
+    }
+
+    let userDataDir = false;
+    const chromiumProfile = this.params.chromiumProfile;
+    if (chromiumProfile) {
+      const isProfileDirAndNotUserData =
+        fs.pathExistsSync(chromiumProfile)
+        && isProfileDir(chromiumProfile)
+        && !isUserDataDir(chromiumProfile);
+
+      if (isProfileDirAndNotUserData) {
+        const pathParts = path.parse(chromiumProfile);
```

Comment:
Let's use the following:

`const { dir: parentDir, base: profileName } = path.parse(chromiumProfile);`

The `parentDir` variable makes it very obvious that we're working with parent directories, which removes the need for the `// now check if the parent dir is a user-data-dir` comment below.

Reply:
like it